### DISCUSSION TITLE
Gatsby Cloud Functions

### DISCUSSION THREAD

**Discussion by tim-line** [gatsbyjs/gatsby#32269] [Help]
We're in the process of moving over a couple of serverless functions from Netlify to Gatsby Cloud and had a couple of questions I couldn't find answers for you in the documentation ... 

1. We use a serverless function for search and write the index to it on createPages based on graphql nodes. From what I can tell Gatsby are packaging up the functions prior to this step (on first run our search results are empty). Is there anyway to change at what point the function gets deployed or work around this?
2. When developing locally I continually get console messages from the 'function watcher' claiming changes are made to my functions when loading a new page. 
3. Similar to the above I have one function which queries a remote API but seems to get called repeatedly when developing locally. Are there any gotchas for this kind of behaviour?

Thanks in advance for any help.

**Comment by KyleAMathews** [comment]
For 1) there's this PR to move compilation later https://github.com/gatsbyjs/gatsby/pull/31840

Not sure what would be causing 2 & 3. I haven't seen that personally or seen any reports. If you can provide a reproduction, that'd help.

**Reply by tim-line** [reply]
Thanks - I'll see if I can isolate the code and provide an example.

Just as note the function causing an issue is pulled in every page. We use it to return the 3 latest articles on useEffect in an attempt to avoid having to render and deploy every page on the site on each publish (this is a large site with over 17000 pages). 

Can you give me any further info on what causes the 'function watcher' to trigger rebuilding the schema and pages when running 'gatsby develop'? I assume a code change to the file would do this but are there any other conditions?
